src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -633,10 +633,9 @@ private void OutputType(CppGenerationBuffer sb, TypeDesc t, bool full)
                 sb.Append("class " + mangledName.Substring(current));
                 if (!t.IsValueType)
                 {
-                    var baseType = t.BaseType;
```

Comment:
To prevent a name collision below now I pulled the VTable stuff out of its null check (since you get back an empty list now).